pkg/webui/lib/download-object-as-file.js

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export default function downloadObjectAsFile(type, object, filename) {
```

Comment:
With the new approach outlined in my other comment, this function will not be needed anymore.

EDIT: Although if you fetch the data only on button click, you might need to look into how to suspend the actual click event until the data response has arrived.